pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/parser/GroovyShellDecoratorImpl.java

```diff
@@ -47,11 +47,12 @@ public void configureCompiler(@CheckForNull final CpsFlowExecution execution, Co
             @Override
             public void call(SourceUnit source, GeneratorContext context, ClassNode classNode) throws CompilationFailedException {
                 boolean doModelParsing = false;
-
                 // First, we'll parse anything that's not coming from an actual source file on disk - i.e., Jenkinsfiles
                 // (which are read in as strings and then compiled), dynamically loaded/evaluated strings, etc. We can
-                // tell when we've got one of those by looking to see if the code source's location is "file:/groovy/shell".
-                if (context.getCompileUnit().getCodeSource().getLocation().toString().equals("file:/groovy/shell")) {
+                // tell when we've got one of those by looking to see if the code source's location is "file:/groovy/shell"
+                // and if the classNode is a script.
+                if (context.getCompileUnit().getCodeSource().getLocation().toString().equals("file:/groovy/shell") &&
+                        classNode.isScript()) {
```

Comment:
Note that this basically means that we'll transform a `Jenkinsfile`, something that's been `evaluate`d, etc, but not any other classes in those files. Before this, we'd end up getting called twice if there was a `Class` in the `Jenkinsfile` as well as the `pipeline` block, and things would blow up good at that point - I'm still not entirely clear how `isDeclarativePipelineStep(it)` is true when `parsePipelineStep(pst, secondaryRun)` (where the first `it` ends up as `pst`) passes the transformed statement, not the original one, but hey, that's what was happening and this makes it not happen.